src/litserve/server.py

```diff
@@ -99,16 +102,31 @@ def run_single_loop(lit_api, request_queue: Queue, request_buffer):
             pipe_s.send(y_enc)
 
 
-def inference_worker(
-    lit_api,
-    device,
-    worker_id,
-    request_queue,
-    request_buffer,
-    max_batch_size,
-    batch_timeout,
-):
+def run_streaming_loop(lit_api, request_queue: Queue, request_buffer):
+    while True:
+        try:
+            uid = request_queue.get(timeout=1.0)
+            try:
+                x_enc, pipe_s = request_buffer.pop(uid)
+            except KeyError:
+                continue
+        except (Empty, ValueError):
+            continue
+
+        x = lit_api.decode_request(x_enc)
+        for y in lit_api.predict(x):
+            y_enc = lit_api.encode_response(y)
```

Comment:
Shouldn't encode_response be aware of streaming here? We may encode differently at the beginning and end of the stream, I think we should make encode_response a generator like predict in the streaming case.

Reply:
you mean something like @lantiga - 

```py
x = lit_api.decode_request(x_enc)
y = lit_api.predict(x)  # returns a generator
y_enc = lit_api.encode_response(y)  # accepts and returns a generator


##################################
def encode_response(y):
    for item in y:
        yield do_something(item)
```